HIVE-23871: ObjectStore should properly handle MicroManaged Table props

pgaref opened the pull request:
ObjectStore should properly handle MicroManaged Table properties

Change-Id: Ia5db047419a11504f3c6047a1eb63acd2a14bdc3

## NOTICE

Please create an issue in ASF JIRA before opening a pull request,
and you need to set the title of the pull request which starts with
the corresponding JIRA issue number. (e.g. HIVE-XXXXX: Fix a typo in YYY)
For more details, please see https://cwiki.apache.org/confluence/display/Hive/HowToContribute

mustafaiman commented on data/files/table1:
```diff
@@ -0,0 +1,5 @@
+1	Acura	4
```
can we give the file a non generic name?

pgaref replied:
sure, done

mustafaiman commented on ql/src/test/results/clientpositive/llap/load_micromanaged_delim.q.out:
```diff
@@ -0,0 +1,186 @@
+NULL	NULL	NULL
+NULL	NULL	NULL
+NULL	NULL	NULL
+NULL	NULL	NULL
```
I did not get the behavior of the last two tables. Why do they print all NULLs?

pgaref replied:
This is exactly the behavior this patch fixes! Apparently I run the test only before the changes -- this is now updated

pgaref commented on ql/src/test/queries/clientpositive/load_micromanaged_delim.q:
```diff
@@ -0,0 +1,32 @@
+SELECT * FROM delim_table_ext;
+
+-- SET hive.create.as.acid=true
+-- SET hive.create.as.insert.only=true
```
Creates the same behaviour as the Table properties below but I agree it makes sense to remove it

pgaref commented:
Thanks for the review @mustafaiman ! Addressed your comments as part of the second commit -- I am still expecting some q.out difference so I am currently waiting for the qtests to run
